src/main/java/app/model/boundary/BoundaryFactory.java

```diff
@@ -18,16 +18,23 @@ public static ArrayList<Boundary> make(SettingsObject obj)
         objects.add(create(obj.getType(), corner[2], corner[3], obj.getTeleportTo()));
         objects.add(create(obj.getType(), corner[3], corner[0], obj.getTeleportTo()));
         objects.removeAll(Collections.singleton(null));
+        if(obj.getType() == FurnitureType.TARGET)
+        {
+            double x = obj.getRect().getMinX() + obj.getRect().getWidth()/2;
+            double y = obj.getRect().getMinY() + obj.getRect().getHeight()/2;
+            objects.add(new Flag(new Vector(x, y), 10));
+        }
         return objects;
     }
 
     private static Boundary create(FurnitureType f, Vector a, Vector b, Vector teleport)
     {
         switch (f)
         {
+            case SHADE, GUARD_SPAWN, INTRUDER_SPAWN -> { return new BoundaryImp(a, b);}
             case WALL, TOWER, BORDER -> { return new VisibleBoundary(a, b);}
             case GLASS -> { return  new TransparentBoundary(a, b);}
-            case SHADE, GUARD_SPAWN, INTRUDER_SPAWN, TARGET -> { return new BoundaryImp(a, b);}
+            case TARGET -> {return new TargetBoundary(a, b);}
```

Comment:
why does it use TargetBoundary here, but Flag in line 25?

Reply:
This is a boundary factory, not a Furniture factory, so it creates a boundary.